src/librustdoc/json/mod.rs

```diff
@@ -1,47 +1,204 @@
+        cache
+            .impls
+            .get(&id)
+            .map(|impls| {
+                impls
+                    .iter()
+                    .filter_map(|i| {
+                        let item = &i.impl_item;
+                        if item.def_id.is_local() {
+                            self.insert(item.clone(), cache);
+                            Some(item.def_id.into())
+                        } else {
+                            None
+                        }
+                    })
+                    .collect()
+            })
+            .unwrap_or_default()
+    }
+}
 
 impl FormatRenderer for JsonRenderer {
     fn init(
-        _krate: clean::Crate,
+        krate: clean::Crate,
         _options: RenderOptions,
         _render_info: RenderInfo,
         _edition: Edition,
         _cache: &mut Cache,
     ) -> Result<(Self, clean::Crate), Error> {
-        unimplemented!()
+        debug!("Initializing json renderer");
+        Ok((JsonRenderer { index: Rc::new(RefCell::new(FxHashMap::default())) }, krate))
     }
 
-    fn item(&mut self, _item: clean::Item, _cache: &Cache) -> Result<(), Error> {
-        unimplemented!()
+    fn item(&mut self, item: clean::Item, cache: &Cache) -> Result<(), Error> {
+        use clean::ItemEnum::*;
+        // Flatten items that recursively store other items by putting their children in the index
+        match item.inner.clone() {
+            StructItem(s) => s.fields.into_iter().for_each(|i| self.insert(i, cache)),
+            UnionItem(u) => u.fields.into_iter().for_each(|i| self.insert(i, cache)),
+            VariantItem(clean::Variant { kind: clean::VariantKind::Struct(v) }) => {
+                v.fields.into_iter().for_each(|i| self.insert(i, cache));
+            }
+            EnumItem(e) => e.variants.into_iter().for_each(|i| self.item(i, cache).unwrap()),
+            TraitItem(t) => t.items.into_iter().for_each(|i| self.insert(i, cache)),
+            ImplItem(i) => i.items.into_iter().for_each(|i| self.insert(i, cache)),
+            _ => {}
+        }
+        self.insert(item.clone(), cache);
+        Ok(())
     }
 
     fn mod_item_in(
         &mut self,
-        _item: &clean::Item,
+        item: &clean::Item,
         _item_name: &str,
-        _cache: &Cache,
+        cache: &Cache,
     ) -> Result<(), Error> {
-        unimplemented!()
+        self.insert(item.clone(), cache);
+        Ok(())
     }
 
     fn mod_item_out(&mut self, _item_name: &str) -> Result<(), Error> {
-        unimplemented!()
+        Ok(())
     }
 
-    fn after_krate(&mut self, _krate: &clean::Crate, _cache: &Cache) -> Result<(), Error> {
-        unimplemented!()
+    fn after_krate(&mut self, krate: &clean::Crate, cache: &Cache) -> Result<(), Error> {
+        debug!("Done with crate");
+        let mut index = (*self.index).clone().into_inner();
+        let trait_items = cache.traits.iter().filter_map(|(id, trait_item)| {
```

Comment:
nit: this could be extracted to a function instead of happening directly in `after_krate`